### ISSUE TITLE
Variables in script.js might get over-written

### ISSUE THREAD

**Issue by duaraghav8** [parkjs814/AlgorithmVisualizer#39]
script.js uses **eval ()** to execute code from data.js & code.js.
**If a user intentionally or un-intentionally creates a variable which is being used by our script.js, that variable gets overwritten.**

Eg-
In the Data Editor,
create variable like:

```
codeEditor = "hello world";
```

Now try running the visualization. The error toast appears because our script **cannot find codeEditor.getValue** property. (Notice line 11 of the Data Editor)

![shot](https://cloud.githubusercontent.com/assets/12758282/15503020/fd502f54-21d4-11e6-9576-ccf302736d4e.png)

Here, try this one more in the Data Editor:

```
dataEditor = "Hello world";  //you will not be able to write "hello world"
```

This is even more damaging. See the effect & you'll understand it.

(See **js/script.js line 196**)
I'm working on a solution for this. Let me know what you think.

**Comment by parkjs814** [comment]
Mhm, that can be a problem. Great point. In which way are you fixing this issue?

**Comment by duaraghav8** [comment]
I'm a bit confused at this point. The simplest way is to use Watch points, but they are not suitable for production.
I'm also looking into [this](http://www.2ality.com/2014/01/eval.html)

The problem is that variables like codeEditor, dataEditor, etc. (who are susceptible) are global. So no matter where you execute eval() from, it WILL overwrite the variables of the same name.

**Comment by parkjs814** [comment]
How do you think about uglifying the code internally before `eval()` it?
Then all the names of variables will have one or two letters, so that the global variables will not be overwritten.

**Comment by duaraghav8** [comment]
You're right! we can do that.
But we can't uglify the variables **tracer** and **G** (in case of Graph algorithms) because these are then being used by script.js. We rely on eval () to create these 2 variables in the global scope so we can use them.

**Comment by parkjs814** [comment]
No worries about it :)
I have removed those dependencies while I implement multiple tracers.
Check out the `dev` branch. (I'm still working on it.)

**Comment by parkjs814** [comment]
I was looking for uglifying library for the client side, but it seems most of them are for node.js.
Do you know any?

**Comment by nem035** [comment]
@parkjs814 take a look at the sources here https://skalman.github.io/UglifyJS-online/
This person made a wrapper around uglify to make it run client-side.

**Comment by duaraghav8** [comment]
This seems fantastic. But its not solving the issue, i.e., all variable names remain intact even after the minification.

**Comment by parkjs814** [comment]
That's not a big deal. We can just avoid short-named variables.
What I'm worrying about is that it could lower the performance of realtime-rendering.

**Comment by parkjs814** [comment]
I figured out that we don't even need to use uglify.
I just moved all the global variables into an anonymous function, and changed all references to it.

fixed at ceadfa439c1b1a2da028d0c35c233f8c7c8e4128
